### PR TITLE
DONOTMERGE Create a windows batch script to run jenkins windows builds.

### PR DESCRIPTION
**Pull request by gunan**

### CODE REVIEW

**Comment by gunan** [comment]
@guschmue @mrry PTAL

**Comment by gunan** [comment]
http://ci.tensorflow.org/job/tensorflow-pr-win-cmake-pytest/7/

Here is the run with this script.
Let's see what happens.

**Comment by guschmue** [comment]
yes, windows build is broken in a couple of ways. The last known good is 832f5fa943dc1f621c29c3194651eb4f688f5fcb.

The most important one (I think that's the one you are seeing) is that 'import tensorflow' can't find the module
tensorflow.python.ops.gen_resource_variable_ops. Something was added and I guess that we need to add it to the cmake pip build. I'll take a look today.
The other issue is that gpu builds are broken: parameterized_truncated_normal_op_gpu.cu.cc (which is new) needs 2 changes to compile on windows. I have a fix for this here https://github.com/guschmue/tensorflow/commits/winbuildfix. 
With that Release for gpu builds again. 
I have not sent a PR yet because RelWithDebugInfo complains about some issue with the pdb for the python wrapper: no permission or disk space but neither is true.
I think something is corrupt in the pdb (like some symbol is too long or so). Have seen this before when I was trying too turn on the normal handling for constexpr in eigen (which is different for windows).
Will try to get a handle on this too but tensorflow.python.ops.gen_resource_variable_ops comes first.

**Comment by vrv** [comment]
@guschmue I just merged a PR that should fix the resource ops.

**Comment by guschmue** [comment]
Yes, that fixed it! Thanks!
Python tests looking good too:
100% tests passed, 0 tests failed out of 141

File: tensorflow/tools/ci_build/windows/cpu/cmake/run_build.bat [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+SET CMAKE_EXE="C:\Program Files\cmake\bin\cmake.exe"
+SET SWIG_EXE="C:\ProgramData\chocolatey\bin\swig.exe"
+SET PY_EXE="C:\Program Files\Anaconda3\python.exe"
+SET PY_LIB="C:\Program Files\Anaconda3\libs\python35.lib"
+
+SET CMAKE_DIR=%REPO_ROOT%\tensorflow\contrib\cmake
+SET MSBUILD_EXE="C:\Program Files (x86)\MSBuild\14.0\Bin\msbuild.exe"
+
+:: Run cmake to create Visual Studio Project files.
+%CMAKE_EXE% %CMAKE_DIR% -A x64 -DSWIG_EXECUTABLE=%SWIG_EXE% -DPYTHON_EXECUTABLE=%PY_EXE% -DCMAKE_BUILD_TYPE=Release -DPYTHON_LIBRARIES=%PY_LIB%
```
Comment (guschmue) [inline tensorflow/tools/ci_build/windows/cpu/cmake/run_build.bat]: add -Dtensorflow_BUILD_PYTHON_TESTS=ON ?

File: tensorflow/tools/ci_build/windows/cpu/cmake/run_py.bat [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+%PY_EXE% -m venv test_venv
+
+:: Activate the virtualenv.
+CALL test_venv\Scripts\activate.bat
+
+:: Turn echo back on, above script turns it off.
+ECHO ON
+
+:: Install the pip package in the virtualenv.
+pip install tf_python\dist\tensorflow-*-py3-none-any.whl
```
Comment (guschmue) [inline tensorflow/tools/ci_build/windows/cpu/cmake/run_py.bat]: I think the * is not going to work in cmd files

File: tensorflow/tools/ci_build/windows/cpu/cmake/run_build.bat (hunk 1 above)
Comment (gunan) [inline tensorflow/tools/ci_build/windows/cpu/cmake/run_build.bat]: Missed this, thanks.

File: tensorflow/tools/ci_build/windows/cpu/cmake/run_py.bat (hunk 2 above)
Comment (gunan) [inline tensorflow/tools/ci_build/windows/cpu/cmake/run_py.bat]: It really does not.
Replacing with hardcoded image name for now.

**Comment by gunan** [comment]
Rebased on top of master again.
Also applied the recommended changes.
Running a new build with this script.

**Comment by gunan** [comment]
OK, here is my test result
http://ci.tensorflow.org/job/tensorflow-pr-win-cmake-pytest/11/console

pip install exits successfully, however ctest runs cannot find tensorflow.
I think we need a virtualenv on windows expert to help us.

**Comment by guschmue** [comment]
without virtenv my jenkins running the tests fine. I can switch this to a virtenv using the same script that you have - will let you know the outcome.

**Comment by guschmue** [comment]
found out why its not working in the virtual env:
in tf_tests.cmake we use:
     add_test(NAME ${sourcefile} COMMAND ${PYTHON_EXECUTABLE} ${sourcefile})

but that PYTHON_EXECUTABLE is not in the virtual env that you install the pip in. 
There are 2 options to fix it:
1. we call cmake and set PYTHON_EXECUTABLE to the python in the virtual env
2. we use 'python' in tf_tests.cmake, than it will take python from $PATH which would be the virtual env

**Comment by gunan** [comment]
In the script, I set PYTHON_EXECUTABLE to python:
(-DPYTHON_EXECUTABLE="python")
Would that not use the python in virtual environment?

**Comment by guschmue** [comment]
cmake will resolve the PYTHON_EXECUTABLE  variable at cmake time. I know there is a -D in ctest but its for some scripting/dashboard mode (I have never used this)

**Comment by gunan** [comment]
Thanks for the pointer.
I will then try to precreate the virtualenv.

**Comment by gunan** [comment]
OK, looks like virtualenv ends up causing more and more issues:
http://ci.tensorflow.org/job/tensorflow-pr-win-cmake-pytest/12/console

The error message:
```
00:14:17.678 ImportError: Traceback (most recent call last):
00:14:17.678   File "C:\tf_jenkins\home\workspace\tensorflow-pr-win-cmake-pytest\cmake_build\test_venv\lib\site-packages\tensorflow\python\__init__.py", line 54, in <module>
00:14:17.678     from tensorflow.python import pywrap_tensorflow
00:14:17.678   File "C:\tf_jenkins\home\workspace\tensorflow-pr-win-cmake-pytest\cmake_build\test_venv\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 28, in <module>
00:14:17.678     _pywrap_tensorflow = swig_import_helper()
00:14:17.678   File "C:\tf_jenkins\home\workspace\tensorflow-pr-win-cmake-pytest\cmake_build\test_venv\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 24, in swig_import_helper
00:14:17.678     _mod = imp.load_module('_pywrap_tensorflow', fp, pathname, description)
00:14:17.678   File "C:\Program Files\Anaconda3\lib\imp.py", line 242, in load_module
00:14:17.678     return load_dynamic(name, filename, file)
00:14:17.678   File "C:\Program Files\Anaconda3\lib\imp.py", line 342, in load_dynamic
00:14:17.678     return _load(spec)
00:14:17.678 ImportError: DLL load failed: The specified module could not be found.
```

Any ideas?

**Comment by guschmue** [comment]
Here is what I use which is working with a virtual env. The only trick is that you need to use always the same python.exe. (the variables that are not defined in the script are set globally in Jenkins, things like 
%SWIG_EXECUTABLE% ...).
Hope that helps.

call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64\vcvars64.bat"
set PreferredToolArchitecture=x64
set rel=Release
set PYTHON_EXECUTABLE=C:\Anaconda3\envs\pyrel\python.exe

cd %WORKSPACE%\tensorflow\contrib\cmake
mkdir build
cd build
cmake .. -A x64 -DCMAKE_BUILD_TYPE=%rel% -DPYTHON_EXECUTABLE=%PYTHON_EXECUTABLE% -DPYTHON_LIBRARIES=%PYTHON_LIBRARIES% -DSWIG_EXECUTABLE=%SWIG_EXECUTABLE% -Dtensorflow_BUILD_PYTHON_TESTS=ON

MSBuild /p:Configuration=%rel% tf_python_build_pip_package.vcxproj

%PYTHON_EXECUTABLE% -m pip install --upgrade %WORKSPACE%\tensorflow\contrib\cmake\build\tf_python\dist\tensorflow-0.11.0rc2_cmake_experimental-py3-none-any.whl
ctest -C %rel%

**Comment by gunan** [comment]
I will close this now and keep working on it.

**Pull request closed without merging**